Assignment 2.Rmd

```diff
@@ -79,22 +85,92 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
 
 #We can change the thickness of the edge according to the number of times a particular teacher has sent a particular student to the principal.
 
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
 plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
 
 ````
 
 ## Part II
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+```{r}
+sum.st <- EDGE %>% group_by(to)%>% summarise(sum(count))
+names(sum.st) = c("id", "count")
+sum.t <- EDGE %>% group_by(from)%>% summarise(sum(count))
+names(sum.t) = c("id", "count")
+total = bind_rows(sum.st, sum.t)
+vertex2 <- full_join(VERTEX, total, by= "id")
+vertex.size = vertex2$count 
+g2 <- graph.data.frame(EDGE, directed=TRUE, vertices=vertex2)
+plot(g2,layout=layout.fruchterman.reingold, vertex.color=vertex2$gender, edge.width=EDGE$count, vertex.size=vertex2$count)
+```
 
 
 
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
+```{r}
+PN<- read.csv(file="hudk4050-classes.csv", header=TRUE, sep=",")
+#clean data'
+PN <- PN[-(1:2),]
+library(dplyr)
+library(tidyr)
+#select useful informationa and unite names
+df <- PN %>% select(Q8,Q9, Q1,Q3,Q4,Q5,Q6,Q7)
+df <- df %>% unite(Name, Q8,Q9, remove = TRUE, sep=" ")
+# gather data
+df1<-gather(df,class_number, class_code, 2:7)
+# delete the messy data
+df2<- df1[-140,]
+df2<-df2[-87,]
+df2<-df2[-34,]
+df2<-df2[-179,]
```

Comment:
This is right. Maybe think about a better way to remove messy data with specific features?